Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -452,10 +451,12 @@ CGImageRef CGImageCreateWithJPEGDataProvider(CGDataProviderRef source,
                                              const CGFloat decode[],
                                              bool shouldInterpolate,
                                              CGColorRenderingIntent intent) {
-    RETURN_NULL_IF((source == nullptr) || ![(NSObject*)source isKindOfClass:[NSData class]]);
+    RETURN_NULL_IF(source == nullptr);
 
-    NSData* sourceData = (__bridge NSData*)source;
-    CGImageRef imageRef = _CGImageLoadJPEG((void*)[sourceData bytes], [sourceData length]);
+    woc::unique_cf<CFDataRef> providerData{ CGDataProviderCopyData(source) };
+    unsigned char* bytes = const_cast<unsigned char*>(CFDataGetBytePtr(providerData.get()));
+    size_t length = CFDataGetLength(providerData.get());
```

Comment:
super duper nit: just pass in them directly rather than having it as a var and then pass it in.